src/libclipper/src/rpc_service.cpp

```diff
@@ -141,16 +145,34 @@ void RPCService::manage_service(const string address) {
       // TODO: Balance message sending and receiving fairly
       // Note: We only receive one message per event loop iteration
       log_info(LOGGING_TAG_RPC, "Found message to receive");
-
       receive_message(socket, connections, connections_containers_map,
                       zmq_connection_id, redis_connection);
+      auto current_time = std::chrono::system_clock::now();
```

Comment:
This logic should be moved outside the `if` conditional. We want to check for dead containers regardless of whether or not a message has been received.